### PR TITLE
FIX-#6897: Preprocess kernel function that alignes columns in groupby

### LINKED ISSUE #6897: Groupby tests on MPI are failing
If you would run:
```
MODIN_ENGINE=unidist MODIN_CPUS=2 mpiexec -n 1 python -m pytest modin/pandas/test/test_groupby.py::test_groupby_sort[True-True] -v
```
You would get this error:
```
modin/pandas/test/test_groupby.py::test_groupby_sort[True-True] Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "miniconda3/envs/modinn/lib/python3.9/site-packages/unidist/api.py", line 92, in init
    init_backend()
  File "miniconda3/envs/modinn/lib/python3.9/site-packages/unidist/core/base/utils.py", line 28, in init_backend
    initialize_mpi()
  File "miniconda3/envs/modinn/lib/python3.9/site-packages/unidist/core/backends/mpi/utils.py", line 12, in initialize_mpi
    init()
  File "miniconda3/envs/modinn/lib/python3.9/site-packages/unidist/core/backends/mpi/core/controller/api.py", line 278, in init
    asyncio.run(worker_loop())
  File "miniconda3/envs/modinn/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "miniconda3/envs/modinn/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "miniconda3/envs/modinn/lib/python3.9/site-packages/unidist/core/backends/mpi/core/worker/loop.py", line 111, in worker_loop
    pending_request = task_store.process_task_request(request)
  File "miniconda3/envs/modinn/lib/python3.9/site-packages/unidist/core/backends/mpi/core/worker/task_store.py", line 423, in process_task_request
    task = request["task"]
KeyError: 'task'

```

### PR DESCRIPTION
**Pull request by dchigarev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [ ] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #6897 
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Comment by dchigarev** [comment]
сс @YarShev

**Review by YarShev — APPROVED** [review]

File: modin/core/dataframe/pandas/dataframe/dataframe.py [hunk 1]
```diff
@@ -3897,7 +3897,9 @@ def compute_aligned_columns(*dfs, initial_columns=None):
                 # aligned columns
                 parts = result._partitions.flatten()
                 aligned_columns = parts[0].apply(
-                    compute_aligned_columns,
+                    # TODO: unidist on MPI execution requires for this function to be preprocessed,
```
Comment (YarShev) [inline modin/core/dataframe/pandas/dataframe/dataframe.py:3900]: We could look at this after https://github.com/modin-project/modin/pull/6900 is merged.

Comment (YarShev) [reply modin/core/dataframe/pandas/dataframe/dataframe.py:3900]: @dchigarev, I ran the previously failing test on latest master without preprocessing `compute_aligned_columns` and it worked. Do you think we should revert this PR? At least, remove the TODO comment.

Comment (dchigarev) [reply modin/core/dataframe/pandas/dataframe/dataframe.py:3900]: Indeed, it seems that #6900 fixed the issue for both dask and unidist.
I'll prepare a PR reverting the change

**Pull request merged**